File: lab1/lab1_sharp/lab1_sharp/ThreadBreaker.cs
using System.Threading;

namespace lab1_sharp
{
    public class ThreadBreaker
    {
        private volatile bool _canBreak;
        private readonly int _sleepTime;

        public ThreadBreaker(int milisecondsToWait)
        {
            _canBreak = false;
            _sleepTime = milisecondsToWait;
        }

        public void Run()
        {
            new Thread(() =>
            {

                Thread.Sleep(_sleepTime);
                _canBreak = true;

            }).Start();
        }

        public bool CanBreak()
        {
            lock (this)
            {
                return _canBreak;
            }
        }
    }
}

File: lab1/lab1_java/src/ThreadBreaker.java
public class ThreadBreaker implements Runnable
{
    private volatile boolean _canBreak;
    private final int _sleepTime;

    public ThreadBreaker(int milisecondsToWait)
    {
        _canBreak = false;
        _sleepTime = milisecondsToWait;
    }

    @Override
    public void run()
    {
        try
        {
            Thread.sleep(_sleepTime);
        } catch (InterruptedException e)
        {
            e.printStackTrace();
        }
        _canBreak = true;
    }

    public synchronized boolean CanBreak()
    {
        return  _canBreak;
    }
}
